vue-components/tests/e2e/specs/Dialog.test.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * Gets the WebElement ID from the result of a `client.activeElement` call.
+ * In chrome,  result.value always has an `ELEMENT` key, whereas in other browser it
+ * has a different format with varying keys. This function assumes that `result.value`
+ * is always an object with a single entry.
+ *
+ * @param {Object} result
+ * @return {string}
+ */
+function getWebElementIdFromActiveElementResult( result ) {
+	return Object.values( result.value )[ 0 ];
+}
+
+describe( 'Dialog', function () {
+
+	beforeEach( function ( client ) {
+		client
+			.pause( 500 )
+			.init( client.launch_url + '/iframe.html?id=dialog--complex&viewMode=story' );
+	} );
+
+	afterEach( function ( client, done ) {
+		client.sauceEnd();// custom command used to finish the test if running in SauceLabs
+		client.end();
+		done();
+	} );
+
+	it( 'traps page focus, so that only visually focused elements are tab-able', function ( client ) {
+
+		const currentBrowser = client.options.desiredCapabilities.browserName;
+
+		// exclude internet explorer for now because there is a known bug in the .keys
+		// and .sendKeys functions
+		// ie: https://github.com/nightwatchjs/nightwatch/issues/2536
+		// firefox: https://github.com/nightwatchjs/nightwatch/issues/2468
+		if ( currentBrowser !== 'internet explorer' && currentBrowser !== 'safari' ) {
+			client
+				.click( '.wikit-Button' )
+				.pause( 500 )
+				.waitForElementPresent( '.wikit-Dialog' )
+				.sendKeys( 'body', client.Keys.TAB )
+				.elementActive( function ( result ) {
+					client.elementIdText( getWebElementIdFromActiveElementResult( result ), ( element ) => {
+						const elementText1 = element.value.replace( /\s+/g, ' ' ).trim();
+						client.assert.equal( elementText1, 'Primary action' );
+					} );
+				} )
+				// sending multiple keys inside array will fail in Firefox.
+				// See: https://github.com/nightwatchjs/nightwatch/issues/2468#issuecomment-738089219
+				.sendKeys( 'body',
+					currentBrowser !== 'firefox' ?
+						client.Keys.TAB + client.Keys.TAB : client.Keys.TAB )
```

Comment:
I am not sure this is solving the same issue though. What happens here is that firefox and non-firefox browsers need a different number of tabs.

Reply:
We would still need to do the conditional within the function. I am also not sure what `sendKeysToActiveElement` is doing. Why does it need the classes?

Reply:
> I am not sure this is solving the same issue though. What happens here is that firefox and non-firefox browsers need a different number of tabs, because they are being counted differently by the browser it seems.

Are you sure? They seem to behave the same when I try it manually in chromium and firefox. My guess is that this is a subtle difference in how the browsers behave when doing `.sendKeys( 'body', ... )`. This `sendKeysToActiveElement` I made is basically a poor man's reimplementation of `.keys()`. Instead of tabbing twice from the `body` element, it finds the currently active element and sends the key to that. Looks like #520 works, so I'll add a clarifying comment to the custom command.